src/lib/reasoners/fun_sat.ml

```diff
@@ -1794,7 +1878,8 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
        inst = Inst.add_predicate env.inst gf dep }
     else
       begin
-        assert (not (ME.mem a_t env.ground_preds));
+        if Stack.is_empty env.incremental.stack_guard then
```

Comment:
pred_def (of ground and non-ground predicates), should take guards into account (see detailed commented in satml_frontend.ml)